src/GetFunctionsMetadataCommand.cs

```diff
@@ -31,25 +32,39 @@ protected override void ProcessRecord()
         {
             try
             {
-                if (!string.IsNullOrEmpty(FunctionsAppDirectory) && Directory.Exists(FunctionsAppDirectory))
+                // Attempt to normalize the user's input to a local path from where the cmdlet is running
+                SessionState ss = new SessionState();
+                string current = Directory.GetCurrentDirectory();
+                Directory.SetCurrentDirectory(ss.Path.CurrentFileSystemLocation.Path);
+                FunctionsAppDirectoryString = Path.GetFullPath(FunctionsAppDirectoryString);
+                Directory.SetCurrentDirectory(current);
+
+                Console.WriteLine(FunctionsAppDirectoryString);
+                if (!Directory.Exists(FunctionsAppDirectoryString))
+                {
+                    ThrowTerminatingError(new ErrorRecord(new Exception(string.Format(AzPowerShellSdkStrings.InvalidFunctionsAppDirectory, FunctionsAppDirectoryString)), "InvalidFunctionAppDirectory", ErrorCategory.ParserError, null));
+                }
+
+                DirectoryInfo FunctionsAppDirectory = new DirectoryInfo(FunctionsAppDirectoryString);
+
+                if(!IsValidFunctionsAppDirectory(FunctionsAppDirectory))
```

Comment:
Please add a test case for this code path as well.

Reply:
Done